src/plot.jl

```diff
@@ -88,49 +88,47 @@ Equal to 0 for undirected graphs. Default: `0.1` for the directed graphs
 Angular width in radians for the arrows. Default: `π/9 (20 degrees)`
 
 `linetype`
-Type of line used for edges ("straight", "curve"). Default: "straight"
+Type of line used for edges (:straight, :curve). Default: :straight
 
 `outangle`
 Angular width in radians for the edges (only used if `linetype = "curve`). 
 Default: `π/5 (36 degrees)`
 
+`backgroundc`
+Color for the plot background. Default: `nothing`
+
 """
 function gplot(g::AbstractGraph{T},
     locs_x_in::Vector{R1}, locs_y_in::Vector{R2};
     nodelabel = nothing,
     nodelabelc = colorant"black",
     nodelabelsize = 1.0,
-    NODELABELSIZE = 4.0,
+    max_nodelabelsize = 4.0,
     nodelabeldist = 0.0,
     nodelabelangleoffset = π / 4.0,
-    edgelabel = [],
+    edgelabel = nothing,
     edgelabelc = colorant"black",
     edgelabelsize = 1.0,
-    EDGELABELSIZE = 4.0,
+    max_edgelabelsize = 4.0,
     edgestrokec = colorant"lightgray",
     edgelinewidth = 1.0,
-    EDGELINEWIDTH = 3.0 / sqrt(nv(g)),
+    max_edgelinewidth = 3.0 / sqrt(nv(g)),
     edgelabeldistx = 0.0,
     edgelabeldisty = 0.0,
     nodesize = 1.0,
-    NODESIZE = 0.25 / sqrt(nv(g)),
+    max_nodesize = 0.25 / sqrt(nv(g)),
     nodefillc = colorant"turquoise",
     nodestrokec = nothing,
     nodestrokelw = 0.0,
     arrowlengthfrac = is_directed(g) ? 0.1 : 0.0,
     arrowangleoffset = π / 9,
-    linetype = "straight",
-    outangle = π / 5) where {T <:Integer, R1 <: Real, R2 <: Real}
-
-    length(locs_x_in) != length(locs_y_in) && error("Vectors must be same length")
-    N = nv(g)
-    NE = ne(g)
-    if nodelabel != nothing && length(nodelabel) != N
-        error("Must have one label per node (or none)")
-    end
-    if !isempty(edgelabel) && length(edgelabel) != NE
-        error("Must have one label per edge (or none)")
-    end
+    linetype = :straight,
+    outangle = π / 5,
+    backgroundc = nothing) where {T <:Integer, R1 <: Real, R2 <: Real}
+
+    @assert length(locs_x_in) == length(locs_y_in) == nv(g) "Position vectors must be of the same length as the number of nodes"
```

Comment:
We should only use `@assert` in situations that we expect that they are always true, as it might be removed by a compiler optimization: https://docs.julialang.org/en/v1/base/base/#Base.@assert

Reply:
I see. I'll use the if -> error form for these.